src/test/ui/rfc-2091-track-caller/std-panic-locations.rs

```diff
@@ -35,4 +38,17 @@ fn main() {
     let fine: Result<(), ()> = Ok(());
     assert_panicked(|| fine.unwrap_err());
     assert_panicked(|| fine.expect_err(""));
+
+    let mut small = [()]; // the implementation backing str, vec, etc
+    assert_panicked(move || { small.index(1); });
+    assert_panicked(move || { small.index_mut(1); });
+
+    let sorted: BTreeMap<bool, bool> = Default::default();
+    assert_panicked(move || { sorted.index(&false); });
+
+    let unsorted: HashMap<bool, bool> = Default::default();
+    assert_panicked(move || { unsorted.index(&false); });
+
+    let weirdo: VecDeque<()> = Default::default();
+    assert_panicked(move || { weirdo.index(1); });
```

Comment:
Can you check the overloaded operator as well? It should be identical, but just in case.